apps/backend/app/agent/providers/llama_index.py

```
@@ -0,0 +1,111 @@
+    modname = provider_name[:dotpos]
+    from importlib import import_module
+    rm = import_module(modname)
+    return getattr(rm, classname), modname, classname
+
+class LlamaIndexProvider(Provider):
+    def __init__(self,
+                 api_key: str = settings.LLM_API_KEY,
+                 api_base_url: str = settings.LLM_BASE_URL,
+                 model_name: str = settings.LL_MODEL,
+                 provider: str = settings.LLM_PROVIDER,
+                 opts: Dict[str, Any] = {}):
```

Ruff (0.12.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)